testing/internal/e2e/tests/gcp/dynamichostcatalog_host_set_empty_test.go

```diff
@@ -0,0 +1,126 @@
+
+	ctx := context.Background()
+	boundary.AuthenticateAdminCli(t, ctx)
+	orgId, err := boundary.CreateOrgCli(t, ctx)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		ctx := context.Background()
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary", e2e.WithArgs("scopes", "delete", "-id", orgId))
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+	projectId, err := boundary.CreateProjectCli(t, ctx, orgId)
+	require.NoError(t, err)
+	hostCatalogId, err := boundary.CreateGCPHostCatalogCli(t, ctx, projectId, c.GcpProjectId, c.GcpClientEmail, c.GcpPrivateKeyId, c.GcpPrivateKey, c.GcpZone)
+	require.NoError(t, err)
+
+	// Set up a host set
+	hostSetId, err := boundary.CreatePluginHostSetCli(t, ctx, hostCatalogId, "labels.empty_test=true")
+	require.NoError(t, err)
+
+	// Check that there are no hosts in the host set
+	t.Logf("Looking for items in the host set...")
+	var actualHostSetCount int
+	for i := 0; i < 3; i++ {
+		if i != 0 {
+			time.Sleep(3 * time.Second)
+		}
+
+		output := e2e.RunCommand(ctx, "boundary",
+			e2e.WithArgs(
+				"host-sets", "read",
+				"-id", hostSetId,
+				"-format", "json",
+			),
+		)
+		require.NoError(t, output.Err, string(output.Stderr))
+		var hostSetsReadResult hostsets.HostSetReadResult
+		err := json.Unmarshal(output.Stdout, &hostSetsReadResult)
+		require.NoError(t, err)
+
+		actualHostSetCount = len(hostSetsReadResult.Item.HostIds)
+		require.Equal(t, 0, actualHostSetCount,
+			fmt.Sprintf("Detected incorrect number of hosts. Expected: 0, Actual: %d", actualHostSetCount),
+		)
+	}
+	t.Log("Successfully detected zero hosts in the host set")
+
+	// Check that there are no hosts in the host catalog
+	t.Logf("Looking for items in the host catalog...")
+	var actualHostCatalogCount int
+	for i := 0; i < 3; i++ {
+		if i != 0 {
+			time.Sleep(3 * time.Second)
+		}
+
+		output := e2e.RunCommand(ctx, "boundary",
+			e2e.WithArgs("hosts", "list", "-host-catalog-id", hostCatalogId, "-format", "json"),
+		)
+		require.NoError(t, output.Err, string(output.Stderr))
+		var hostCatalogListResult hostcatalogs.HostCatalogListResult
+		err := json.Unmarshal(output.Stdout, &hostCatalogListResult)
+		require.NoError(t, err)
+
+		actualHostCatalogCount = len(hostCatalogListResult.Items)
+		require.Equal(t, 0, actualHostCatalogCount,
+			fmt.Sprintf("Detected incorrect number of hosts. Expected: 0, Actual: %d", actualHostCatalogCount),
+		)
+	}
+	t.Log("Successfully detected zero hosts in the host catalog")
+
+	// Create target
+	targetId, err := boundary.CreateTargetCli(t, ctx, projectId, c.GcpTargetPort)
+	require.NoError(t, err)
+	err = boundary.AddHostSourceToTargetCli(t, ctx, targetId, hostSetId)
+	require.NoError(t, err)
+
+	// Attempt to connect to target
+	output := e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"connect",
+			"-target-id", targetId,
+			"-format", "json",
+			"-exec", "/usr/bin/ssh", "--",
+			"-l", c.GcpTargetSshUser,
+			"-i", c.GcpTargetSshKey,
```

Comment:
Does this work? Based on the other test, this contains the key contents? Do we need to create a file and specify the file here?

Reply:
Actually I have to pass the key as a file. I'll fix that. The target connection should still. I'll fix it and test it again